src/services/codefixes/fixClassIncorrectlyImplementsInterface.ts

```diff
@@ -0,0 +1,52 @@
+/* @internal */
+namespace ts.codefix {
+    registerCodeFix({
+        errorCodes: [Diagnostics.Class_0_incorrectly_implements_interface_1.code],
+        getCodeActions: (context: CodeFixContext) => {
+            const sourceFile = context.sourceFile;
+            const start = context.span.start;
+            const token = getTokenAtPosition(sourceFile, start);
+            const checker = context.program.getTypeChecker();
+
+            const classDecl = getContainingClass(token);
+            if (!classDecl) {
+                return undefined;
+            }
+
+            const startPos: number = classDecl.members.pos;
+
+            const implementedTypeNodes = getClassImplementsHeritageClauseElements(classDecl);
+            const implementedTypes = implementedTypeNodes.map(checker.getTypeFromTypeReference);
+            const implementedTypeSymbols = implementedTypes.map(checker.getPropertiesOfType);
+
+            const result: CodeAction[] = [];
+
+            for (const symbols of implementedTypeSymbols) {
+                const symbolMap = createMap<Symbol>();
+                for (const symbol of symbols) {

```

Comment:
i do not think you need this map. the array is generated from a map anyways, so no need to re do the deduplication.